lib/chai-as-promised.js

```diff
@@ -119,6 +119,7 @@
             );
 
             chaiAsPromised.transferPromiseness(that, derivedPromise);
+            return that;
```

Comment:
Nice catch.
Just summarizing this just to make other reviewer's job easier: this works because when we have a `return` inside the getter function then [this `if` condition](https://github.com/chaijs/chai/blob/c273846593dd6cb324106f242f2933f1294523b8/lib/chai/utils/addProperty.js#L48) becomes `true` and therefore we return that instead of a new assertion with flags copied over.